### PR TITLE
[15.0][MIG] mail: delete duplicates which trigger issue due to the new constaint mail_c…

### PR DESCRIPTION
**Pull request by jcdrubay**
…hannel_partner_partner_unique

Fixes: https://github.com/jcdrubay/OpenUpgrade/issues/2

### CODE REVIEW

File: openupgrade_scripts/scripts/mail/15.0.1.5/pre-migration.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+
```
Comment (MiquelRForgeFlow) [inline openupgrade_scripts/scripts/mail/15.0.1.5/pre-migration.py]:
```suggestion
```

File: openupgrade_scripts/scripts/mail/15.0.1.5/pre-migration.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+from openupgradelib import openupgrade
+
+_logger = logging.getLogger(__name__)
+
+
+@openupgrade.migrate(use_env=False)
+def migrate(cr, version):
+    # Cleanup potential duplicates
+    openupgrade.logged_query(
+        cr,
```
Comment (MiquelRForgeFlow) [inline openupgrade_scripts/scripts/mail/15.0.1.5/pre-migration.py]:
```suggestion
@openupgrade.migrate()
def migrate(env, version):
    # Cleanup potential duplicates
    openupgrade.logged_query(
        env.cr,
```
This is @pedrobaeza preference and he has long convinced me to do this way as default.

Comment (pedrobaeza) [reply openupgrade_scripts/scripts/mail/15.0.1.5/pre-migration.py]: You never know when you are going to need the environment, so better to have it already populated for not having a great diff later.

**Comment by jcdrubay** [comment]
@legalsylvain I think that you can remove the label Blockedbydependency

**Comment by legalsylvain** [comment]
Base is not done akaik ?

**Comment by jcdrubay** [comment]
My bad, you are right. I have been dreaming someone had done it. :|

**Comment by jcdrubay** [comment]
This commit was my dream becoming reality.
https://github.com/OCA/OpenUpgrade/pull/2967/commits/130d038040ce522208e14fb921f35f8b4a5aef15

Do I misunderstand something?

**Comment by StefanRijnhart** [comment]
This PR looks a bit messy. Can you try to  rebase?

**Comment by jcdrubay** [comment]
I think it is better to handle here: https://github.com/OCA/OpenUpgrade/pull/3273/files#r899917133

**Pull request closed without merging**